tests/PHPStan/Analyser/data/preg_match_shapes.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace PregMatchShapes;
+
+use function PHPStan\Testing\assertType;
+
+function doMatch(string $s): void {
+	if (preg_match('/Price: (£|€)\d+/', $s, $matches)) {
+		assertType('array{0: string, 1?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+
+	if (preg_match('/Price: (£|€)(\d+)/i', $s, $matches)) {
+		assertType('array{0: string, 1?: string, 2?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+
+	if (preg_match('  /Price: (£|€)\d+/ i u', $s, $matches)) {
+		assertType('array{0: string, 1?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+
+	if (preg_match('(Price: (£|€))i', $s, $matches)) {
+		assertType('array{0: string, 1?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+
+	if (preg_match('_foo(.)\_i_i', $s, $matches)) {
+		assertType('array{0: string, 1?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+
+	if (preg_match('/(a)(b)*(c)(d)*/', $s, $matches)) {
+		assertType('array{0: string, 1?: string, 2?: string, 3?: string, 4?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+}
+
+function doNonCapturingGroup(string $s): void {
+	if (preg_match('/Price: (?:£|€)(\d+)/', $s, $matches)) {
+		assertType('array{0: string, 1?: string}', $matches);
+	}
+	assertType('array<string>', $matches);
+}
+
```

Comment:
It is possible to turn off auto capturing of non named groups with the 'n' flag and modifier:

```suggestion

function doNonAutoCapturingFlag(string $s): void {
    if (preg_match('/(\d+)/n', $s)) {
        assertType('array{0: string}', $s);
    }

    assertType('array{}', $s);
}

function doNonAutoCapturingModifier(string $s): void {
    if (preg_match('/(?n)(\d+)/', $s)) {
        assertType('array{0: string}', $s);
    }

    assertType('array{}', $s);
}

```

Reply:
adjusted and added the test, thanks